### PR TITLE
MessageDB: remove TaskSeq dep

### PR DESCRIPTION
**Pull request by nordfjord**
It's simply not justifiable to have a dependency on the library for this one little thing that doesn't benefit from it 😅

### CODE REVIEW

File: src/Equinox.MessageDb/MessageDb.fs [hunk 1]
```diff
@@ -198,20 +196,28 @@ module Read =
             "Read", count, slice.LastVersion)
         return slice }
 
-    let private readBatches (log: ILogger) (readSlice: int64 -> int -> ILogger -> CancellationToken -> Task<StreamEventsSlice>)
-            (maxPermittedBatchReads: int option) (startPosition: int64) ct
-        : taskSeq<int64 * ITimelineEvent<EventBody>[]> =
-        let rec loop batchCount pos: taskSeq<int64 * ITimelineEvent<EventBody>[]> = taskSeq {
+    let private readBatches (log : ILogger) batchSize (readSlice : int64 -> int -> ILogger -> CancellationToken -> Task<StreamEventsSlice>)
```
Comment (nordfjord) [inline src/Equinox.MessageDb/MessageDb.fs:199]:
```suggestion
    let private readBatches (log: ILogger) batchSize (readSlice: int64 -> int -> ILogger -> CancellationToken -> Task<StreamEventsSlice>)
```

File: src/Equinox.MessageDb/MessageDb.fs [hunk 2]
```diff
@@ -198,20 +196,28 @@ module Read =
...
+            (maxPermittedBatchReads : int option) (startPosition : int64) ct
```
Comment (nordfjord) [inline src/Equinox.MessageDb/MessageDb.fs:200]:
```suggestion
            (maxPermittedBatchReads: int option) (startPosition: int64) ct
```

File: src/Equinox.MessageDb/MessageDb.fs [hunk 3]
```diff
@@ -198,20 +196,28 @@ module Read =
...
-            yield slice.LastVersion, slice.Messages
+            let! slice = readSlice pos batchCount batchLog ct |> Async.AwaitTaskCorrect
+            version <- max version slice.LastVersion
+            result.AddRange(slice.Messages)
             if not slice.IsEnd then
-                yield! loop (batchCount + 1) (slice.LastVersion + 1L) }
-        loop 0 startPosition
+              batchCount <- batchCount + 1
+              pos <- slice.LastVersion  + 1L
+              do! loop () }
```
Comment (nordfjord) [inline src/Equinox.MessageDb/MessageDb.fs:217]:
```suggestion
                batchCount <- batchCount + 1
                pos <- slice.LastVersion  + 1L
                do! loop () }
```

**Comment by bartelink** [comment]
Very nice work, thanks for pushing on this

**Pull request merged**